src/command-line/upgrade.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+const colors = require("chalk");
+const program = require("commander");
+const Helper = require("../helper");
+const Utils = require("./utils");
+
+program
+	.command("upgrade")
+	.option("-p, --package [name]", "Update only the specified package")
+	.description("Update installed themes and packages to their latest versions. Specify one theme or package to update with the --package argument.")
+	.on("--help", Utils.extraHelp)
+	.action(function(cmd) {
+		const fs = require("fs");
+		const path = require("path");
+
+		log.info(`Updating ${(cmd.package !== undefined ? colors.green(cmd.package) : "packages")}...`);
+
+		// Get paths to the location of packages directory
+		const packagesPath = Helper.getPackagesPath();
+		const packagesConfig = path.join(packagesPath, "package.json");
+		const argsList = [
+			"upgrade",
+			"--latest",
+			"--json",
+			"--production",
+			"--ignore-scripts",
+			"--non-interactive",
+			"--cwd",
+			packagesPath,
+		];
+
+		// Check if the configuration file exists
+		if (!fs.existsSync(packagesConfig)) {
+			log.error("There are no packages installed.");
+			return;
+		}
+
+		// If a package name is supplied, check it exists
+		if (cmd.package !== undefined) {
+			if (JSON.parse(fs.readFileSync(packagesConfig)).dependencies.hasOwnProperty(cmd.package)) {
+				log.error(`${colors.green(cmd.package)} is not installed.`);
+				return;
+			}
+
+			argsList.splice(1, 0, cmd.package);
+		}
+
+		return Utils.executeYarnCommand.apply(
+			this,
+			argsList
```

Comment:
I believe `Utils.executeYarnCommand(...argsList)` does exactly this and is a bit nicer than the `.apply(this, ...)` pattern.

(This is kinda why I preferred the argument of `executeYarnCommand` to be one array instead of a dynamic number of arguments)